### PR TITLE
auth-issue

### PR DESCRIPTION
**Pull request by Vuyo-Ngwane**
Key Updates:

User Registration and Login:

Implemented the registerUser function to allow new users to create accounts with hashed passwords.
Implemented the loginUser function to authenticate users, generating JWT tokens for valid credentials.
Admin Functions:

Added getAllUsers and deleteUser functions in the adminController to enable admin users to manage the user database.
Middleware for authentication ensures that only admin users can access sensitive routes.
User Routes:

Created dedicated routes for user registration and login in user.js.
Added routes for managing user ingredients, including fetching and adding ingredients.
Error Handling:

Enhanced error handling across all functions to provide clear feedback for different error scenarios, such as invalid credentials and internal server errors.
Routing Updates:

Organized routes in admin.js and user.js to streamline the API structure.
This refactor enhances the application's user management capabilities while ensuring security through token-based authentication and access control.

### CODE REVIEW

**Review by Vuyo-Ngwane — COMMENTED** [review]
Resolved conflicts on index.js and added updated authRoutes.js

File: backend-app/server/routes/authRoutes.js
Comment (ImisebenziEmihle) [inline backend-app/server/routes/authRoutes.js:1]: Hey mate, please check line 10, we're missing a '/' on the register route.

Comment (ImisebenziEmihle) [inline backend-app/server/routes/authRoutes.js:1]: Hey mate, please check line 10, we're missing '/' from the register route, on authRoutes.js file.

File: backend-app/server/controllers/userController.js
Comment (ImisebenziEmihle) [inline backend-app/server/controllers/userController.js:1]: @Vuyo-Ngwane please check line 11, all new users register with their username, email and password..  Please add the email on there too. the role by default is User.

File: backend-app/server/routes/user.js
Comment (ImisebenziEmihle) [inline backend-app/server/routes/user.js:1]: @Vuyo-Ngwane check line 18 to 24 in this user.js. The code should exist in admin.js , it is an /admin route. All admin routes must be called from admin.js

**Review by NokulungaM — APPROVED** [review]
Good work!

**Comment by Vuyo-Ngwane** [comment]
Awesome

**Pull request merged**